Source/WTF/wtf/JSONValues.cpp

```diff
@@ -646,36 +725,30 @@ void Value::writeJSON(StringBuilder& output) const
             output.append(m_value.number);
         break;
     }
-    default:
+    case Type::Object:
+    case Type::Array:
         ASSERT_NOT_REACHED();
     }
 }
 
 size_t Value::memoryCost() const
+{
+    return visitDerived([&](auto& value) {
+        return value.memoryCostImpl();
+    });
+}
+
+size_t Value::memoryCostImpl() const
 {
     size_t memoryCost = sizeof(this);
```

Comment:
This looks wrong. Shouldn’t it be `sizeof(*this)`?

Reply:
Looks right. Changed.